daemon/p2p_announce.c

```diff
@@ -12,6 +12,24 @@
 #include <ccan/tal/tal.h>
 #include <secp256k1.h>
 
+struct queued_message {
+	int type;
+
+	/* Unique tag specifying the msg origin */
+	void *tag;
+
+	/* Timestamp for `channel_update`s and `node_announcement`s, 0
+	 * for `channel_announcement`s */
+	u32 timestamp;
+
+	/* Serialized payload */
+	u8 *payload;
+
+	struct list_node list;
+
+	/* who told us about this message? */
+	struct peer *origin;
+};
```

Comment:
Do we want origin?  I think if peer goes away, these messages stay around, risking this becoming a dangling pointer?  If we really want it, probably better to stash the id her?